diff --git a/dist/server/dev/middleware-turbopack.js b/dist/server/dev/middleware-turbopack.js
index 09f7711b2e96ce259cdfee6ba8ab9673b2cbf3a1..497af8f737e9831076b39c90a4347ae39d6e1184 100644
--- a/dist/server/dev/middleware-turbopack.js
+++ b/dist/server/dev/middleware-turbopack.js
@@ -289,7 +289,7 @@ function getOverlayMiddleware({ project, projectPath, isSrcDir }) {
             let openEditorResult;
             if (isAppRelativePath) {
                 const relativeFilePath = searchParams.get('file') || '';
-                const appPath = _path.default.join('app', isSrcDir ? 'src' : '', relativeFilePath);
+                const appPath = _path.default.join(isSrcDir ? 'src' : '', 'app', relativeFilePath);
                 openEditorResult = await (0, _launcheditor.openFileInEditor)(appPath, 1, 1, projectPath);
             } else {
                 const frame = createStackFrame(searchParams);
diff --git a/dist/server/dev/middleware-webpack.js b/dist/server/dev/middleware-webpack.js
index 3dccc88289ed87b6618abb88c157079366f521fc..64e73dece3c5cb98db01c69bb26f55a7defe3b15 100644
--- a/dist/server/dev/middleware-webpack.js
+++ b/dist/server/dev/middleware-webpack.js
@@ -407,7 +407,7 @@ function getOverlayMiddleware(options) {
             const isAppRelativePath = searchParams.get('isAppRelativePath') === '1';
             if (isAppRelativePath) {
                 const relativeFilePath = searchParams.get('file') || '';
-                const appPath = _path.default.join('app', isSrcDir ? 'src' : '', relativeFilePath);
+                const appPath = _path.default.join(isSrcDir ? 'src' : '', 'app', relativeFilePath);
                 openEditorResult = await (0, _launcheditor.openFileInEditor)(appPath, 1, 1, rootDirectory);
             } else {
                 // TODO: How do we differentiate layers and actual file paths with round brackets?
